### PR TITLE
[Card#37]Update leave local and remote

### PR DESCRIPTION
**Pull request by HarshwardhanSingh**
Trello card: https://trello.com/c/T2LenbVq

### CODE REVIEW

File: app/controllers/leaves_controller.rb [hunk 1]
```diff
@@ -37,17 +37,34 @@ def create
...
+      end_date = params[:end].to_time(:utc).iso8601
+      summary = "#{current_user.name} is on Leave (#{Reason.find_by(id: params[:leave][:reason_id]).name})"
+    end
     leave = {
-      summary: params[:summary],
-      start: {dateTime: params[:start].to_time(:utc).iso8601},
-      end: {dateTime: params[:end].to_time(:utc).iso8601}
+      summary: summary,
+      start: {dateTime: start_date},
+    end: {dateTime: end_date}
```
Comment (deepak) [inline app/controllers/leaves_controller.rb]: not indented properly ?

File: app/controllers/leaves_controller.rb [hunk 2]
```diff
@@ -37,17 +37,34 @@ def create
...
     }
-    @client.update_leave(params[:calendar_id], params[:id], leave)
+    calendarsleaves = Calendarleave.where(leave_id: params[:leave_id])
```
Comment (deepak) [inline app/controllers/leaves_controller.rb]: weird naming. can call it `calendar_leaves`

File: app/controllers/leaves_controller.rb [hunk 3]
```diff
@@ -37,17 +37,34 @@ def create
...
+    calendarsleaves.each do |calendarleave|
+      calendar = Calendar.find(calendarleave.calendar_id)
```
Comment (deepak) [inline app/controllers/leaves_controller.rb]: why is this query for `Calendar` inside a loop ?
does it change on each iteration or is it constant ?

Comment (HarshwardhanSingh) [inline app/controllers/leaves_controller.rb]: @deepak Leave can belong to multiple calendars so in that case we have to find all the calendars and pass their google_id to delete_leave method to update that leave on all calendars.

File: app/controllers/leaves_controller.rb [hunk 4]
```diff
@@ -37,17 +37,34 @@ def create
...
     leave = {
-      summary: params[:summary],
-      start: {dateTime: params[:start].to_time(:utc).iso8601},
-      end: {dateTime: params[:end].to_time(:utc).iso8601}
+      summary: summary,
+      start: {dateTime: start_date},
+      end: {dateTime: end_date}
     }
-    @client.update_leave(params[:calendar_id], params[:id], leave)
+    calendar_leaves = Calendarleave.where(leave_id: params[:leave_id])
```
Comment (deepak) [inline app/controllers/leaves_controller.rb]: N + 1 query. use includes for eager loading

File: app/controllers/leaves_controller.rb [hunk 5]
```diff
@@ -37,17 +37,34 @@ def create
...
+    calendar_leaves.each do |calendarleave|
+      calendar = Calendar.find(calendarleave.calendar_id)
+      @client.update_leave(calendar.google_id, calendarleave.leave_google_id, leave)
+    end
+    leave_db = Leave.find(params[:leave_id])
```
Comment (deepak) [inline app/controllers/leaves_controller.rb]: move finds on params on top as much as possible. want to fail early if the params are invalid. not after running the bigger calendar-leaves + calendar query

**Pull request merged**